app/src/main/java/com/mad41/tripreminder/Registration_Form.java

```diff
@@ -0,0 +1,134 @@
+package com.mad41.tripreminder;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class Registration_Form extends AppCompatActivity {
+    EditText firstName, email,password,confirmationPassword;
+    FirebaseAuth firebaseAuth;
+    ProgressBar progressBar;
+    Button Signup ;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_registration__form);
+        firstName = (EditText)findViewById(R.id.userName);
+        email = (EditText)findViewById(R.id.E_mail);
+        password = (EditText)findViewById(R.id.passwordID);
+        Signup = findViewById(R.id.signUp);
+        confirmationPassword = (EditText)findViewById(R.id.ConfirmPasswordID);
+        progressBar = findViewById(R.id.signUPprogressBar);
+        firebaseAuth = FirebaseAuth.getInstance();
+        Signup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signUp();
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+    }
+   private void UpdateUi(FirebaseUser firebaseUser) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.